device/iot-device-client/pom.xml

```diff
@@ -82,14 +83,43 @@
             <artifactId>commons-codec</artifactId>
             <version>1.10</version>
         </dependency>
+        <!-- Logging support -->
+        <dependency>
+   			<groupId>org.slf4j</groupId>
+ 			<artifactId>slf4j-api</artifactId>
+    	</dependency>
         <dependency>
             <groupId>log4j</groupId>
             <artifactId>log4j</artifactId>
             <version>1.2.17</version>
+            <scope>runtime</scope>
```

Comment:
Yes. Otherwise the test classes won't work. They need a logger implementation. But dependency to log4j should not be populated to users of the SDK.